src/vision/status_quo/alan_finds_database_drops_hard.md

```diff
@@ -0,0 +1,104 @@
+
+```rust
+use sqlx::Connection;
+
+#[async_std::main]
+async fn do_the_thing() -> Result<(), sqlx::Error> {
+    // Create a connection
+    let conn = SqliteConnection::connect("sqlite::memory:").await?;
+
+    // Make a simple query to return the given parameter
+    let row: (i64,) = sqlx::query_as("SELECT $1")
+        .bind(150_i64)
+        .fetch_one(&conn).await?; // <----- if this await is cancelled, doesn't help
+
+    assert_eq!(row.0, 150);
+    
+    // he adds this:
+    log!("closing the connection");
+    conn.close();
+
+    Ok(())
+}
+```
+  
+* He observes that in the cases where he has the problem the log statement never executes.
+* He asks Barbara for help and she explains how `await` can be canceled and it will the destructors for things that are in scope.
+* He reads the [source for the SqliteConnection destructor](https://github.com/launchbadge/sqlx/blob/0ed524d65c2a3ee2e2a6706910b85bf2bb72115f/sqlx-core/src/pool/connection.rs#L70-L74) and finds that destructor spawns a task to actually close the connection.
+* He realizes there is a race condition:
+    * the task may not have actually closed the connection before `do_the_thing` is called a second time
+* He gives up and gets frustrated
+
+Next, Alan seeks verification and validation of his understanding of the source code from the `sqlx` forum. His advisor states that Rust doesn't have a way to execute async operations in a destructor.
+
+## Finding the Solution
+
+Alan briefly considers rearchitecting his application in more extreme ways to retain use of async, but he gives up and seeks a more straight forward solution. He discovers `rusqlite`, a sychronous database library and adopts it. This requires some rearchitecting but solves the problem.
+
+## 🤔 Frequently Asked Questions
+
+*Here are some standard FAQ to get you started. Feel free to add more!*
+
+* **What are the morals of the story?**
+    * The story demonstrates solid research steps that Alan uses to understand and resolve his problem.
+    * Completion of the [Cancellation and timeouts docs](https://rust-lang.github.io/async-book/06_multiple_futures/01_chapter.html) may have been helpful. It's difficult to know how something absent might have improved the solution search process.
+    * Stating the obvious, with more async experience Alan would shorten his solution search process.
+* **What are the sources for this story?**
+    * Sergey Galich reported this problem, among many others.
+* **Why did you choose Alan to tell this story?**
+    * His experience and understanding of other languages coupled with his desire to apply Rust would likely lead him to try solutions before deeply researching them.
+* **How would this story have played out differently for the other characters?**
+    * It is likely that Nickalas would have reached out to Barbara or possibly the sqlx community sooner. On the other hand Barbara, given her deep experience in Rust wouldn't have selected async in her initial design.
```

Comment:
I'm not sure this part about Barbara is true. I would've probably tried to use async at first. This seems like a pretty subtle bug that could easily take a while for anyone to understand.

Reply:
I'd probably just say that it might well have played out similarly for everyone.

Reply:
change committed